pkg/policyengine/connectivitypdp/connectivity_pdp_test.go

```diff
@@ -73,7 +73,7 @@ func TestPrivilegedVsRegular(t *testing.T) {
 	require.Equal(t, connectivitypdp.DefaultDenyPolicyName, decisions[0].MatchedBy)
 	require.Equal(t, false, decisions[0].PrivilegedMatch)
 
-	err = pdp.AddOrUpdatePolicy(&trivialConnPol, false)
+	err = pdp.AddOrUpdatePolicy(connectivitypdp.PolicyFromCRD(&trivialConnPol))
```

Comment:
would you be able to hide the internal rep if the code accepted `interface{}` (or using generics) and then you internally call the right function (privileged or not) based on type assertion?